packages/react-storage/src/components/StorageBrowser/Views/LocationActionView/UploadControls.tsx

```diff
@@ -200,7 +218,11 @@ const parseSelectionData = (
 export const UploadControls = (): JSX.Element => {
   const [{ history, path }] = useControl({ type: 'NAVIGATE' });
   const [files, setFiles] = React.useState<File[]>([]);
-  const [fileSelect, handleSelect] = useFileSelect(setFiles);
+  const [fileSelect, handleSelect] = useFileSelect((newFiles) => {
+    setFiles((prevFiles) => {
+      return mergeSelectedFiles(prevFiles, newFiles);
+    });
+  });
```

Comment:
While this helps add new files and folders into the current table, if a user removes a task or multiple tasks and tries to add new files/folders, the tasks that were previously removed get added back onto the table because the `task.cancel` does not update the `files` state

Does anyone have thoughts on this current implementation?